homeassistant/components/google_assistant/trait.py

```diff
@@ -153,6 +153,8 @@
 COMMAND_LOCATE = f"{PREFIX_COMMANDS}Locate"
 COMMAND_CHARGE = f"{PREFIX_COMMANDS}Charge"
 
+SERVICE_CALL_LIMIT = 2  # Wait 2 seconds for services to finish
```

Comment:
I think we should at least double it?

Reply:
This is a really good comment. I'm not sure what's best.

Testing Google Assistant with a Tasmota switch, which relies on pushed state updates from the devices:
```
2021-11-18 11:42:09 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event google_assistant_command[L]: request_id=17999020846578459602, entity_id=['switch.tasmota_2'], execution=[{'command': 'action.devices.commands.OnOff', 'params': {'on': True}}], source=cloud>
2021-11-18 11:42:09 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=switch, service=turn_on, service_data=entity_id=switch.tasmota_2>
2021-11-18 11:42:09 DEBUG (MainThread) [homeassistant.components.google_assistant.smart_home] Result: [{'ids': ['switch.tasmota_2'], 'status': 'SUCCESS', 'states': {'online': True, 'on': False}}]
2021-11-18 11:42:09 DEBUG (MainThread) [hass_nabucasa.iot] Publishing message:
{'msgid': 'f4ca95ed39138c83d9ac27ced98067c0',
 'payload': {'payload': {'commands': [{'ids': ['switch.tasmota_2'],
                                       'states': {'on': False, 'online': True},
                                       'status': 'SUCCESS'}]},
             'requestId': '17999020846578459602'}}

2021-11-18 11:42:09 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=switch.tasmota_2, old_state=<state switch.tasmota_2=off; friendly_name=Tasmota @ 2021-11-18T11:41:39.595294+01:00>, new_state=<state switch.tasmota_2=on; friendly_name=Tasmota @ 2021-11-18T11:42:09.313408+01:00>>
```

We miss the state update, and reply with the old state.
However, the google home app shows the switch as on, it doesn't revert back to off.
Maybe Google don't really care about the state update?

Somehow, I think we should do either of:
- Call the service with a very short timeout, and don't worry about missing state updates
- Call the service with a very short timeout, and then wait some time for state update from the entity - this means we would be able to report an updated state for entities which get push updates